### ISSUE TITLE
tcpdump with online status guest

### ISSUE THREAD

**Issue by ahroobe** [cuckoosandbox/cuckoo#2362]
Thanks for creating an issue! But first: did you read our community guidelines?
https://cuckoo.sh/docs/introduction/community.html

##### My issue is:

Host : Ubuntu 64bit
Guest : Windows 7 64bit

I want to sniff all the packets from guest(Windows 7 64bit).
but with only host-only adapter(vboxnet0), network is not working and network results have only udp packets(multicast).

I added NAT to guest machine, but I can't find the way to get packets through NAT.

and then, changed NAT to bridged adapter(adapter name ens33), I checked packets from guest is caught by tcpdump (host machine).
then I changed interface to ens33(in virtualbox.conf), but network result was empty(even udp packets!)

How can I get the TCP packets from guest machine with internet?
 
##### My Cuckoo version and operating system are:

2.0.4

**Comment by RicoVZ** [comment]
Hi ahroobe,

Thanks for posting an issue. :smile: 

The interface field in your machinery config should be `vboxnet0` in your case. The most likely reason you were/are only getting UDP is because there are no forwarding rules configured that allow your VMs to communicate with the internet.

An example of global rules that allow all hosts on the subnet on the interface vboxnet0 can be found here https://cuckoo.sh/docs/installation/host/routing.html#simple-global-routing.

Cuckoo also allows for per-analysis routing. For more information, see this docs page: https://cuckoo.sh/docs/installation/host/routing.html#per-analysis-network-routing-options

**Comment by ahroobe** [comment]
Thank you for noticing me that site.
I set guest PC(windows) got only one network adapter (host-only adapter, vboxnet0) and tried iptables commands. 
But I don't understand how this host-only network can be connected to the internet. Inside, only ping to 192.168.56.1 works because it's host-only network (result shows only UDP packet).
I added NAT to second adapter but it still doesn't work. It can ping to 8.8.8.8 but all packets to out are not caught by cuckoo.  
I'm sorry, but can you explain more details about the network settings of virtualbox?

below is ifconfig of host machine.
![image](https://user-images.githubusercontent.com/12726078/42608202-b8e333d6-85c1-11e8-9cea-0544bbac9467.png)

I typed commands(in simple-global-routing) as follows.
![image](https://user-images.githubusercontent.com/12726078/42608283-06a9b842-85c2-11e8-9dbe-6033c80281ae.png)

**Comment by RicoVZ** [comment]
@ahroobe The iptables rules mentioned here: https://cuckoo.sh/docs/installation/host/routing.html#simple-global-routing

I see you tried them, you have some typo's in them, however. This is also why iptables shows the 'bad argument' errors.

>But I don't understand how this host-only network can be connected to the internet. Inside, only ping to 192.168.56.1 works because it's host-only network (result shows only UDP packet).

Well, you are right. It is not connected to the internet. In short: if you add the iptables rules, it tells your operating system to forward the traffic from the `vboxnet0 subnet` interface to your `ens33` output interface and  forward the replies back to `vboxnet0 subnet`.

So, assuming your interface and vboxnet0 subnet on that screenshot is your actual VM subnet, and you do not have any other conflicting forwarding rules, the rules you can apply are:

```
sudo iptables -t nat -A POSTROUTING -o ens33 -s 192.168.56.0/24 -j MASQUERADE
sudo iptables -P FORWARD DROP
sudo iptables -A FORWARD -m state --state RELATED,ESTABLISHED -j ACCEPT
sudo iptables -A FORWARD -s 192.168.56.0/24 -j ACCEPT
sudo iptables -A FORWARD -s 192.168.56.0/24 -d 192.168.56.0/24 -j ACCEPT
```

To flush all existing rules and set all policies to accept, use these rules. **This deletes all existing rules**.

```
sudo iptables -P INPUT ACCEPT
sudo iptables -P FORWARD ACCEPT
sudo iptables -P OUTPUT ACCEPT
sudo iptables -t nat -F
sudo iptables -t mangle -F
sudo iptables -F
sudo iptables -X
```

**Comment by ahroobe** [comment]
I followed instructions as you suggested and I succeeded in watching the outgoing packets in tcpdump. However, reports.json still has only UDP packets results. (Internet in guest machine still does not work as well)
I'm really sorry that I'm not familiar with the routing rules and keep doing something wrong.

I'd rather use bridged network for analysis and change sniffing interface to ens33, though it is dirty.

I think it can be done by changing the code in cuckoo/auxiliary/sniffer.py
~~~
pargs = [
            tcpdump, "-U", "-q", "-s", "0", "-n",
            "-i", self.machine.interface,
        ]
~~~
to
~~~
pargs = [
            tcpdump, "-U", "-q", "-s", "0", "-n",
            "-i", "ens33",
        ]
~~~

or change interface in virtualbox.conf, but it doesn't work as I mentioned in the first question. (I don't understand why) 

~~~
# (Optional) Specify the name of the network interface that should be used
# when dumping network traffic from this machine with tcpdump. If specified,
# overrides the default interface specified in auxiliary.conf
# Example (vboxnet0 is the interface name):
interface = ens33 
~~~

Is it right to do this?

**Comment by doomedraven** [comment]
You should use host only mode and vboxnet0, that why you may have problems

**Comment by RicoVZ** [comment]
@ahroobe 
>However, reports.json still has only UDP packets results. (Internet in guest machine still does not work as well)

Did you change the interface field in `virtualbox.conf` back to vboxnet0? :smile: 

Also, I just thought of this: forwarding should be enabled on your interface `vboxnet0`.
You can do that by using the following command:
`echo 1 | sudo tee -a  /proc/sys/net/ipv4/conf/vboxnet0/forwarding`.

**Comment by ahroobe** [comment]
No, I didn't change it,,, but it works now!
I referenced another website https://unix.stackexchange.com/questions/383791/virtualbox-host-only-with-internet/384187#384187 to fix it.

after flush iptables rules, I typed 
~~~
sudo sysctl -w net.ipv4.ip_forward=1
sudo iptables -t filter -I FORWARD --in-interface vboxnet0 --out-interface ens33 --source 192.168.56.0/24 -j ACCEPT
iptables -t filter -I FORWARD --in-interface ens33 --out-interface vboxnet0 --destination 192.168.56.0/24 -j ACCEPT
iptables -t nat -I POSTROUTING -o ens33 -j MASQUERADE
~~~

iptables rules are below
![image](https://user-images.githubusercontent.com/12726078/42804214-436008d4-89e3-11e8-8175-0b9752966ed1.png)

Thank you all!
